rust/osm2lanes-web/src/main.rs

```diff
@@ -120,14 +175,64 @@ impl Component for App {
 }
 
 impl App {
+    fn calculate(value: &String) -> Result<(Lanes, Tags), Result<(Lanes, String), String>> {
+        log::trace!("Calculate: {}", value);
+        match Tags::from_str(&value) {
+            Ok(tags) => match get_lane_specs_ltr_with_warnings(&tags, &CFG) {
+                Ok(lanes) => match lanes_to_tags(&lanes.lanes, &CFG) {
+                    Ok(tags) => Ok((lanes, tags)),
+                    Err(e) => Err(Ok((lanes, e.to_string()))),
+                },
+                Err(e) => Err(Err(e.to_string())),
+            },
+            Err(_) => Err(Err("parsing tags failed".to_owned())),
```

Comment:
Wee nesting! Reasonable

Reply:
Yeah, Maybe Ill do the right thing in the future and ternary enum it, but it worked :P